src/parallel/StaticIrregDomainDecomposition.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file StaticIrregDomainDecompostion.cpp
+ * @author amartyads
+ * @date 21.02.24
+ */
+
+#include "StaticIrregDomainDecomposition.h"
+#include "Domain.h"
+#include "utils/Logger.h"
+#include <fstream>
+#include <sstream>
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(Domain* domain) : 
+	StaticIrregDomainDecomposition(domain, MPI_COMM_WORLD, {std::vector<unsigned int>{},{},{}}) {}
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(
+	Domain* domain, MPI_Comm comm, const std::array<std::vector<unsigned int>, DIMgeom>& subdomainWeights) : 
+		DomainDecomposition(comm, {(int)subdomainWeights[0].size(), (int)subdomainWeights[1].size(), (int)subdomainWeights[2].size()}), 
+		_subdomainWeights(subdomainWeights),
+		_boxMin{0},
+		_boxMax{0},
+		_domainLength{domain->getGlobalLength(0), domain->getGlobalLength(1), domain->getGlobalLength(2)} {
+			if(_subdomainWeights[0].size() == 0 && _subdomainWeights[1].size() == 0 && _subdomainWeights[2].size() == 0) { //default behaviour, regular grid
+				for (int i = 0; i < 3; i++) {
+					for (int j = 0; j < _gridSize[i]; j++) {
+						_subdomainWeights[i].push_back(1);  //equal subdomains, so weight = 1
+					}
+				}
```

Comment:
- [ ] add comments what the loops are for
- [ ] reserve before you push back in a loop